packages/server/lib/browsers/memory/index.ts

```diff
@@ -221,12 +228,20 @@ export const calculateMemoryStats: () => Promise<void> = measure(async () => {
   const maxAvailableRendererMemory = Math.min(jsHeapSizeLimit, currentAvailableMemory + rendererProcessMemRss)
 
   const rendererUsagePercentage = (rendererProcessMemRss / maxAvailableRendererMemory) * 100
-  // if we're using more than MEMORY_THRESHOLD_PERCENTAGE of the available memory,
+  // if the renderer's memory is above the MEMORY_THRESHOLD_PERCENTAGE, we should collect garbage on the next test
   const shouldCollectGarbage = rendererUsagePercentage >= MEMORY_THRESHOLD_PERCENTAGE && !SKIP_GC
 
   // if we should collect garbage, set the flag to true so we can collect garbage on the next test
   collectGarbageOnNextTest = collectGarbageOnNextTest || shouldCollectGarbage
 
+  // if the renderer's memory is above the EMERGENCY_MEMORY_THRESHOLD_PERCENTAGE, we should perform an emergency garbage collection now
+  const shouldEmergencyCollectGarbage = rendererUsagePercentage >= EMERGENCY_MEMORY_THRESHOLD_PERCENTAGE && !SKIP_GC
+
+  if (shouldEmergencyCollectGarbage) {
+    debug('emergency garbage collection triggered')
+    await checkMemoryPressure(automation, shouldEmergencyCollectGarbage)
```

Comment:
Does this reset `shouldCollectGarbage`?

Reply:
No, I debated whether or not I should clear it, and decided not to. Thoughts?

Reply:
If we don't reset, we could be collecting garbage on the next test even if the threshold is below 50 percent right?
The next cycle of checking would enable collect garbage again if we didn't drop below 50% right?

Reply:
RIGHT? sorry i said right too many times in my last comment 😅

Reply:
Right :) Though if it didn't drop below threshold and the next test started prior to the next interval we would be missing a GC. I decided to err on the side of more GCs.

Reply:
🤷‍♂️, fair point.